### PR TITLE
raft-canonical: 0.9.6 -> 0.9.17, dqlite: 1.0.0 -> 1.4.0, lxd: 3.18 -> 3.22

### PR DESCRIPTION
**Pull request by wucke13**
###### Motivation for this change
Update.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by wucke13** [comment]
@GrahamcOfBorg build raft-canonical

**Comment by wucke13** [comment]
@GrahamcOfBorg build lxd dqlite raft-canonical

**Comment by wucke13** [comment]
@GrahamcOfBorg build dqlite
@GrahamcOfBorg build raft-canonical

**Comment by wucke13** [comment]
@GrahamcOfBorg lxd

**Comment by FRidh** [comment]
What's the status? I see ofborg reported a failure.

**Comment by wucke13** [comment]
@GrahamcOfBorg build raft-canonical dqlite lxd
@FRidh Probably finished, let me run a last test

**Comment by wucke13** [comment]
@GrahamcOfBorg build raft-canonical dqlite lxd

File: pkgs/tools/admin/lxd/default.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+, raft-canonical, sqlite-replication, udev
 , writeShellScriptBin, apparmor-profiles, apparmor-parser
 , criu
 , bash
 }:
 
 buildGoPackage rec {
   pname = "lxd";
-  version = "3.18";
+  version = "3.21";
```
Comment (marsam) [inline pkgs/tools/admin/lxd/default.nix]: 3.22 was recently released, would you mind updating it? https://discuss.linuxcontainers.org/t/lxd-3-22-has-been-released/7027

Comment (wucke13) [reply pkgs/tools/admin/lxd/default.nix]: Sure, wouldn't be the first time on this PR...

Comment (wucke13) [reply pkgs/tools/admin/lxd/default.nix]: Done!

**Comment by wucke13** [comment]
@GrahamcOfBorg build lxd

**Comment by marsam** [comment]
@wucke13 thank you for working on this, please ping me if a PR gets stuck

**Pull request merged**